__init__.py

```diff
@@ -246,7 +247,7 @@ def __add_key_and_cookie_file(self,
 
             cookie_file = self.cookie_file or expand_paths(osx_cookies,'osx')
 
-        elif sys.platform.startswith('linux'):
+        elif sys.platform.startswith('linux') or search('bsd', sys.platform).group(0):
```

Comment:
@3Gigs 

It will cause an `AttributeError: 'NoneType' object has no attribute 'group'` when `sys.platform` is neither Linux nor BSD. 

Why not use `'bsd' in sys.platform`? It is faster than using regex.

Reply:
Sorry I'm not really that familiar with Python, but wouldn't ```'bsd' in sys.platform``` mean that sys.platform would have to be exactly 'bsd'? There are multiple derivatives of bsd and I would like to support all of them, which is why I used regex.

Reply:
@3Gigs 
`'bsd' in sys.platform` will do exactly what you wanted.
If the string contains the 'bsd' substring the value will be true

For example:
```python
>> 'bsd' in 'freebsd'
True
>> 'bsd' in 'openbsd'
True
>> 'bsd' in 'dragonfly bsd'
True
>> 'bsd' in 'linux'
False
```
N.B.: The substring searching is case-sensitive. So I would recommend using `sys.platform.lower()`